src/net.cpp

```diff
@@ -1119,6 +1138,9 @@ void CConnman::DisconnectNodes()
                 // remove from m_nodes
                 m_nodes.erase(remove(m_nodes.begin(), m_nodes.end(), pnode), m_nodes.end());
 
+                // Tell the eviction manager to forget about this node
+                m_evictionman.RemoveCandidate(pnode->GetId());
```

Comment:
I think this could/should be done outside of `m_nodes_mutex` for the sake of untangling? I'm not actually sure, but if it wouldn't break anything IMO it'd be preferable.

We'd need to create a list of nodes to be evicted, then do the removals outside of the lock.

Reply:
Done.

(You actually had it like that in your original branch and i changed it because it seemed simpler this way.)